spec/cictl/index_command_spec.rb

```diff
@@ -2,17 +2,63 @@
 
 require "spec_helper"
 require "cictl/deleted_records"
+require "cictl/journal"
 
 RSpec.describe CICTL::IndexCommand do
   before(:each) do
     CICTL::SolrClient.new.empty!.commit!
     ENV["CICTL_ZEPHIR_FILE_TEMPLATE_PREFIX"] = "sample"
+    init_journals
   end
 
   after(:each) do
     CICTL::SolrClient.new.empty!.commit!
     ENV.delete "CICTL_ZEPHIR_FILE_TEMPLATE_PREFIX"
     remove_test_log
+    init_journals
+  end
+
+  describe "#index continue" do
+    context "with no journal" do
+      it "indexes all example records" do
+        update_file_count = CICTL::Examples::EXAMPLES.count { |ex| ex[:type] == :upd }
+        CICTL::Commands.start(["index", "continue", "--quiet", "--log", test_log])
+        expect(solr_count).to eq CICTL::Examples.all_ids.count
+        expect(Dir.children(HathiTrust::Services[:journal_directory]).count).to eq(update_file_count)
+      end
+    end
+
+    context "with only the full file" do
+      it "indexes only the update files and writes a journal for each" do
+        CICTL::Examples::EXAMPLES.select { |ex| ex[:type] == :full }
+          .each do |ex|
+            CICTL::Journal.new(date: Date.parse(ex[:date]), full: ex[:type] == :full).write!
```

Comment:
I'm waffling a bit on how much logic we should have in the tests vs. just spelling out what files to write or to expect to be present. I do think that `CICTL::Examples` is helpful in keeping things organized, but maybe if in CICTL::Examples we specifically added what the expected journal filename was rather than computing it here, and considered adding a `type` helper there to get examples by type? For example if we were able to write:

```ruby
CICTL::Examples.type(:full).each do |ex|
  FileUtils.touch(ex.journal_file)
end
update_file_count = CICTL::Examples.type(:upd).count
update_ids = CICTL::Examples.type(:upd,:delete).map(&:ids).flatten
```
here and

```ruby
CICTL::Examples.type(:full,:upd).each do |ex|
  FileUtils.touch(ex.journal_file)
end
```

that might be a good compromise and make the tests a little easier to follow?

Reply:
Yup, I have done a variation on the suggested changes. Same intent, slightly different window dressing.